mysql-binlog-stream-shared/app/io/laserdisc/mysql/json/package.scala

```diff
@@ -0,0 +1,52 @@
+package io.laserdisc.mysql
+
+import io.circe._
+import io.circe.parser._
+
+package object json {
+  def flatHash(doc: String, removeKey: String = ""): Either[Exception, String] =
+    flatten(doc) match {
+      case Left(e) => Left(e)
+      case Right(js) =>
+        Right(
+          if (removeKey == "")
+            md5sum(js.mkString(","))
+          else
+            md5sum(js.filter(!_.startsWith(removeKey)).mkString(","))
+        )
+    }
+
+  def flatten(doc: String): Either[Exception, List[String]] =
+    parse(doc) match {
+      case Left(e)     => Left(e)
+      case Right(json) => Right(flatSort(json))
+    }
+
+  def flatSort(json: Json): List[String] = flatUnsorted(json).sorted
+
+  def flatUnsorted(json: Json): List[String] =
```

Comment:
mind adding a quick doc to each of these?

Reply:
it flattens the json tree and sorts the result